### PR TITLE
Add GitHub Actions workflow to run CI.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Travis CI is becoming increasingly unavailable. This ports the same test suite to run via GitHub Actions based on the workflow I used for rosdep. It does add additional operating systems and python versions to the test matrix.

### CODE REVIEW

**Comment by clalancette** [comment]
Should we just remove the `.travis.yml` from here while we are at it?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            python: 2.7
+          - os: ubuntu-18.04
+            python: 3.6
+      name: rosinstall_generator tests
+      runs-on: ${{matrix.os}}
+
+      steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{matrix.python}}
+        uses: actions/setup-python@v1
```
Comment (cottsay) [inline .github/workflows/ci.yaml:26]: Any reason not to use v2 of this action?
```suggestion
        uses: actions/setup-python@v2
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: rosinstall-generator-ci
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: ['*']
```
Comment (cottsay) [inline .github/workflows/ci.yaml:7]: I'm still not convinced that this works as intended.

```suggestion
```

**Review by cottsay — APPROVED** [review]
A couple of non-critical items, but LGTM.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (nuclearsandwich) [reply .github/workflows/ci.yaml]: Nope, just making minimal changes to stale PRs and didn't notice.

**Pull request merged**